### PR TITLE
Preserve ordering of collections in traversableWrites

### LINKED ISSUE #95: toJson does not always preserve the order of elements in collection
### Play Version 2.6.3

I found that `LowPriorityWrites.traversableWrites` does not always preserve the order.
For instance we have `scala.collection.mutable.SortedSet` of elements, however json has wrong output because of `as.map(w.writes(_))`
I suppose it should be something like `as.toSeq.map(w.writes(_)`

### Expected Behavior

```
Json.toJson(scala.collection.mutable.SortedSet(1, 2, 3)) = [1, 2, 3]
```

### Actual Behavior

```
Json.toJson(scala.collection.mutable.SortedSet(1, 2, 3)) = [2,1,3]
```

### PR DESCRIPTION
**Pull request by gmethvin**
Fixes #95

### CODE REVIEW

File: play-json/shared/src/test/scala/WritesSharedSpec.scala [hunk 1]
```diff
@@ -23,6 +23,16 @@ class WritesSharedSpec extends WordSpec with MustMatchers {
     }
   }
 
+  "Traversable Writes" should {
+    "write Seqs" in {
+      Json.toJson(Seq(5, 4, 3, 2, 1)) mustEqual Json.arr(5, 4, 3, 2, 1)
+    }
+    "write SortedSets" in {
+      import scala.collection.immutable.SortedSet
+      Json.toJson(SortedSet(1, 2, 3, 4, 5)) mustEqual Json.arr(1, 2, 3, 4, 5)
```
Comment (cchantep) [inline play-json/shared/src/test/scala/WritesSharedSpec.scala]: `immutable.SortedSet` is already ok with master for me, would rather test with a mutable one

File: play-json/shared/src/main/scala/Writes.scala [hunk 2]
```diff
@@ -301,10 +301,16 @@ sealed trait LowPriorityWrites extends EnvWrites {
...
    * Serializer for Traversables types.
    */
-  implicit def traversableWrites[A: Writes] = {
+  implicit def traversableWrites[A: Writes]: Writes[Traversable[A]] = {
     val w = implicitly[Writes[A]]
 
-    Writes[Traversable[A]] { as => JsArray(as.map(w.writes(_)).toArray[JsValue]) }
+    Writes[Traversable[A]] { as =>
+      val builder = mutable.ArrayBuilder.make[JsValue]()
+      for (a <- as) {
```
Comment (cchantep) [inline play-json/shared/src/main/scala/Writes.scala]: `as.foreach`? As each iteration is side-effect

File: play-json/shared/src/test/scala/WritesSharedSpec.scala (hunk 1 above)
Comment (gmethvin) [reply play-json/shared/src/test/scala/WritesSharedSpec.scala]: This test fails on master. I just added another one with the mutable version for completeness though.

File: play-json/shared/src/main/scala/Writes.scala (hunk 2 above)
Comment (gmethvin) [reply play-json/shared/src/main/scala/Writes.scala]: This doesn't have a `yield` so it is just side effecting, and desugars to `foreach`. That said, there is is no real advantage in readability here for `for` vs `foreach` so I'll switch to foreach.

**Comment by gmethvin** [comment]
@cchantep Updated

**Pull request merged**